### ISSUE TITLE
[FEATURE] Ice autoban filter

### ISSUE THREAD

**Issue by miffi** [unixporn/trup-rs#77]
Honestly ice is so predictable now that we could just humor him and code an autoban into the bot to kick him out.

Things to check for ice:
- Pings pome, aurelian, mander and sometimes, wooosh
- Account creation date less than 3 days old
- Set list of swear words known to be used by him in conjunction in pings

**Comment by JavaCafe01** [comment]
I don't think we should add specific code just for dealing with Ice. Maybe we can have the bot auto-ban anyone who is new and gets muted for spamming (or says something from the blocklist).

**Comment by elkowar** [comment]
Someone give me a concrete set of conditions as to _when_ to autoban someone, then sure! should definitely not be super ice-specific, but just autoban anyone who spams or does other stupid shit immediately after joining, imo

**Comment by mlvzk** [comment]
I think muting and deleting their messages would be better than banning. At least then you can contact mods in case of misfire. Also let's you be more lenient with the algorithm.

**Comment by iSparsh** [comment]
I think what should be done is:-
1. Mute for 10(?) mins if mass spam. Mutes for 30 mins if the account is new (>= 3 days) and spams
2. If they get unmuted and spam again, within a set amount of time, they get muted for longer (eg: infraction-break time: 30mins; punishment: 1 hour)
3. The mute time gets longer and longer and the infraction-breaking time threshold can get shorter/longer whatever needs to be implemented.
4. 3 chances and you're banned.

We could create a custom role for people who spam and have a new account to have permanent slow-mode but that would be too much hassle.
Anything is flexible here, of course, but imo we shouldn't be auto banning.

**Comment by mTvare6** [comment]
[Ice sees this github issue too you can't stop him](https://github.com/unixporn/robbb/issues/77)
